src/libraries/Common/src/System/TimeProvider.cs

```
@@ -10,115 +10,100 @@ namespace System
         }
 
         /// <summary>
         /// Gets a <see cref="DateTimeOffset"/> value whose date and time are set to the current
         /// Coordinated Universal Time (UTC) date and time and whose offset is Zero,
         /// all according to this <see cref="TimeProvider"/>'s notion of time.
         /// </summary>
-        public abstract DateTimeOffset UtcNow { get; }
+        /// <remarks>
+        /// The default implementation returns <see cref="DateTimeOffset.UtcNow"/>.
+        /// </remarks>
+        public virtual DateTimeOffset GetUtcNow() => DateTimeOffset.UtcNow;
 
         private static readonly long s_minDateTicks = DateTime.MinValue.Ticks;
         private static readonly long s_maxDateTicks = DateTime.MaxValue.Ticks;
 
         /// <summary>
         /// Gets a <see cref="DateTimeOffset"/> value that is set to the current date and time according to this <see cref="TimeProvider"/>'s
-        /// notion of time based on <see cref="UtcNow"/>, with the offset set to the <see cref="LocalTimeZone"/>'s offset from Coordinated Universal Time (UTC).
+        /// notion of time based on <see cref="GetUtcNow"/>, with the offset set to the <see cref="LocalTimeZone"/>'s offset from Coordinated Universal Time (UTC).
         /// </summary>
-        public DateTimeOffset LocalNow
+        public DateTimeOffset GetLocalNow()
         {
-            get
-            {
-                DateTime utcDateTime = UtcNow.UtcDateTime;
-                TimeSpan offset = LocalTimeZone.GetUtcOffset(utcDateTime);
-
-                long localTicks = utcDateTime.Ticks + offset.Ticks;
-                if ((ulong)localTicks > (ulong)s_maxDateTicks)
-                {
-                    localTicks = localTicks < s_minDateTicks ? s_minDateTicks : s_maxDateTicks;
-                }
+            DateTimeOffset utcDateTime = GetUtcNow();
+            TimeSpan offset = (LocalTimeZone ?? TimeZoneInfo.Local).GetUtcOffset(utcDateTime);
```

This trying to protect against a derived type erroneously returning null from LocalTimeZone?  If we want to protect against that, I think we should throw an exception instead... otherwise it suggests returning null is acceptable.